validator/java/.gitignore

```diff
@@ -0,0 +1,9 @@
+# These directories get generated by Maven.
+target/
+src/main/resources/
+src/main/java/dev/amp/validator/ValidatorProtos.java
+bazel-bin
+bazel-java
+bazel-out
+bazel-testlogs
+bin
```

Comment:
For now, I've collapsed all the bazel directories here with `bazel-*`, so this is only 5 now. The bits in `src/main/**` have to be in the paths they are in order for the Java tests to run properly, so that leaves `target`, `bazel-*`, and `bin`. For the time being, I'd like to punt on reducing this file further, as it seems more manageable now with at least the bazel wildcard.

Reply:
SGTM.